src/app/organizations/state/register-organization.service.ts

```diff
@@ -0,0 +1,55 @@
+import { HttpErrorResponse } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { MatDialog } from '@angular/material';
+import { Router } from '@angular/router';
+
+import { AlertService } from '../../shared/alert/state/alert.service';
+import { Organisation, OrganisationsService } from '../../shared/swagger';
+import { FormsState } from '../registerOrganization/register-organization.component';
+
+@Injectable({ providedIn: 'root' })
+export class RegisterOrganizationService {
+
+  readonly urlRegex = /https?:\/\/(www\.)?[-a-zA-Z0-9@:%._\+~#=]{2,256}\.[a-z]{2,4}\b([-a-zA-Z0-9@:%_\+.~#?&//=]*)/;
```

Comment:
rather than rolling your own, there may be a library for this already (hopefully included as a dependency already)
something like https://validatejs.org/#validators-url  or https://developer.mozilla.org/en-US/docs/Web/API/URL/URL